be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+    void filterBySelectorAssert(vectorized::IColumn::MutablePtr col, std::vector<uint16_t> selector,
+                                const IDataType& dt, MutableColumnPtr should_sel_col,
+                                size_t expect_size) {
+        // only used in column_nullable and predict_column, column_dictionary
+        EXPECT_TRUE(col->is_nullable() || col->is_column_dictionary() ||
+                    col->is_predicate_column());
+        // for every data type should assert behavior in own UT case
+        col->clear();
+        col->insert_many_defaults(should_sel_col->size());
+        std::cout << "col size:" << col->size() << std::endl;
+        auto sel = should_sel_col->clone_empty();
+        Status st = col->filter_by_selector(selector.data(), expect_size, sel);
+        EXPECT_EQ(st, Status::OK());
+        EXPECT_EQ(sel->size(), expect_size);
+        printColumn(*sel, dt);
+    }
+
+    void assertPermutationsWithLimit(const IColumn::Permutation& lhs,
+                                     const IColumn::Permutation& rhs, size_t limit) {
+        if (limit == 0) {
+            limit = lhs.size();
+        }
+
+        for (size_t i = 0; i < limit; ++i) {
+            ASSERT_EQ(lhs[i], rhs[i]);
+        }
+    }
+
+    // this function is common function to produce column, according to ColumnValueGetter
+    // this range_size can be set, and it is helpfully make common column data which can be inserted into columns.
+    template <typename ColumnValueGetter>
+    void generateRanges(std::vector<std::vector<Field>>& ranges, size_t range_size,
+                        ColumnValueGetter getter) {
+        for (auto& range : ranges) {
+            range.clear();
+        }
+
+        size_t ranges_size = ranges.size();
+
+        for (size_t range_index = 0; range_index < ranges_size; ++range_index) {
+            for (size_t index_in_range = 0; index_in_range < range_size; ++index_in_range) {
+                auto value = getter(range_index, index_in_range);
+                ranges[range_index].emplace_back(value);
+            }
+        }
+    }
+
+    void insertRangesIntoColumn(std::vector<std::vector<Field>>& ranges,
+                                IColumn::Permutation& ranges_permutations,
+                                vectorized::IColumn& column) {
+        for (const auto& range_permutation : ranges_permutations) {
+            auto& range = ranges[range_permutation];
+            for (auto& value : range) {
+                column.insert(value);
+            }
+        }
+    }
+
+    // this function helps to check sort permutation behavior for column
+    void stableGetColumnPermutation(const IColumn& column, bool ascending, size_t limit,
+                                    int nan_direction_hint, IColumn::Permutation& out_permutation) {
+        (void)(limit);
```

Comment:
What's this?

Reply:
to get column permutation in order by compare_at() function which can make a sequence  result for   ```column.get_permutation```